pkg/graveler/graveler.go

```diff
@@ -1493,15 +1494,18 @@ func (g *Graveler) GCNewRunID() string {
 	return g.garbageCollectionManager.NewID()
 }
 
-func (g *Graveler) GetBranchProtectionRules(ctx context.Context, repository *RepositoryRecord) (*BranchProtectionRules, string, error) {
+func (g *Graveler) GetBranchProtectionRules(ctx context.Context, repository *RepositoryRecord) (*BranchProtectionRules, *string, error) {
 	return g.protectedBranchesManager.GetRules(ctx, repository)
 }
 
 func (g *Graveler) SetBranchProtectionRules(ctx context.Context, repository *RepositoryRecord, rules *BranchProtectionRules, lastKnownChecksum *string) error {
 	if lastKnownChecksum == nil {
 		return g.protectedBranchesManager.SetRules(ctx, repository, rules)
 	}
-	return g.protectedBranchesManager.SetRulesIf(ctx, repository, rules, *lastKnownChecksum)
+	if *lastKnownChecksum == "" {
```

Comment:
Seem like the logic should be reversed?:
1. lastKnownChecksum is nil -> SetIf
2. lastKnownChecksum is "" -> Set
And of course change the logic in the invoking methods accordingly

Reply:
Agree that this is ugly. I argue that "" as the "skip validation checksum" is just as problematic. Added a dedicated skip-validation constant to overcome this. This will be removed once the legacy API is dropped.